packages/sdk/src/identity/IdentityMapping.ts

```diff
@@ -1,63 +1,68 @@
 import { MLDSAKeyPairIdentity } from './MLDSAKeyPairIdentity'
 import { SignatureType } from '@streamr/trackerless-network'
+import { ECDSAKeyPairIdentity } from './ECDSAKeyPairIdentity'
+import { KEY_TYPES } from '@streamr/utils/dist/src/signingUtils'
 
 /**
  * This is where config keyTypes are connected to Identity implementations.
  * 
  * How to configure new Identity types:
  * 1. Add a new SignatureType entry to NetworkRpc.proto in network package
- * 2. Add the needed SigningUtil to signingUtils.ts
- * 3. Create the Identity implementation itself (eg. extend KeyPairIdentity)
+ * 2. Add the needed SigningUtil to signingUtils.ts and tests to signingUtils.test.ts
+ * 3. Add the new SigningUtil to exports.ts in the utils package
+ * 3. Create the Identity implementation itself (eg. extend KeyPairIdentity) and tests for it
  * 4. Wire everything together below
  */
-export const validKeyTypeValues = ['ECDSA_SECP256K1_EVM', 'ML_DSA_87'] as const
-/** @internal */
-export const IdentityMapping: Record<KeyType, {
+export const IDENTITY_MAPPING: {
+    keyPairType: KeyType
+    // Used by createIdentityFromConfig
     fromConfig: (config: Pick<StrictStreamrClientConfig, 'auth'>) => Identity
-    generate: () => Identity
-    signingUtil: SigningUtil
+    // Used by SignatureValidator
     signatureType: SignatureType
-}> = {
-    'ECDSA_SECP256K1_EVM': { 
+}[] = [
+    { 
+        keyPairType: 'ECDSA_SECP256K1_EVM',
         fromConfig: EthereumKeyPairIdentity.fromConfig, 
-        generate: EthereumKeyPairIdentity.generate,
-        signingUtil: ECDSA_SECP256K1_EVM,
         signatureType: SignatureType.ECDSA_SECP256K1_EVM,
     },
-    'ML_DSA_87': {
+    { 
+        keyPairType: 'ECDSA_SECP256R1',
+        fromConfig: ECDSAKeyPairIdentity.fromConfig, 
+        signatureType: SignatureType.ECDSA_SECP256R1,
+    },
+    {
+        keyPairType: 'ML_DSA_87',
         fromConfig: MLDSAKeyPairIdentity.fromConfig,
-        generate: MLDSAKeyPairIdentity.generate, 
-        signingUtil: ML_DSA_87,
         signatureType: SignatureType.ML_DSA_87,
-    }
-}
+    },
+] as const
 
-export type KeyType = typeof validKeyTypeValues[number]
 export const DEFAULT_KEY_TYPE: KeyType = 'ECDSA_SECP256K1_EVM'
 
 // Static check that all valid key types have corresponding factory functions above
-validKeyTypeValues.forEach((keyType) => {
-    if (!(keyType in IdentityMapping)) {
-        throw new Error(`Missing factory function for keyType: ${keyType}`)
+KEY_TYPES.forEach((keyType) => {
+    if (!IDENTITY_MAPPING.find((id) => id.keyPairType === keyType)) {
+        throw new Error(`keyType missing from IDENTITIES: ${keyType}`)
     }
 })
 
 /**
  * Creates an Identity instance based on what's in the StreamrClient config
  */
-export const createIdentityFromConfig = (config: Pick<StrictStreamrClientConfig, 'auth' | 'contracts'>): Identity => {
+export function createIdentityFromConfig(config: Pick<StrictStreamrClientConfig, 'auth' | 'contracts'>): Identity {
```

Comment:
Just stylistic change?

Reply:
Yeah, do we prefer one style over the other?